phobos_umss.sh

```diff
@@ -0,0 +1,490 @@
+    return $error_7
+}
+
+#######################
+# Secondary functions #
+#######################
+
+# Outputs a log line in stdout, or in $LOGFILE if any.
+save_log() {
+    if [ $LEVEL -eq $DEBUG ]; then
+        timestamp=$(date +"%Y:%m:%d-%T.%N")
+        # Messages building
+        if [ "$2" ]; then
+            message_colour="${Black}${ID} ${Green}[${timestamp}] ${Red}${1}(${Blue}${2}${Red})${Color_Off}"
+            message_black="${ID} [${timestamp}] ${1}(${2})"
+        else
+            message_colour="${Black}${ID} ${Red}${1}${Color_Off}"
+            message_black="${ID} ${1}"
+        fi
+
+        # Message(s) writing on file(s)
+        if [ -n "${LOGFILE_FANCY+x}" ]; then
+            if [ "$LOGFILE_FANCY" != "" ]; then
+                echo -e "$message_colour" >> "$LOGFILE_FANCY"
+            fi
+        fi
+
+        if [ -n "${LOGFILE_SOBRE+x}" ]; then
+            if [ "$LOGFILE_SOBRE" != "" ]; then
+                echo -e "$message_black" >> "$LOGFILE_SOBRE"
+            fi
+        fi
+
+        # Message(s) displaying on screen
+        if [ $DISPLAY == 1 ]; then
+            if [ $DISPLAY_COLOUR == 1 ]; then
+                echo -e "$message_colour"
+            fi
+            if [ $DISPLAY_BLACK == 1 ]; then
+                echo -e "$message_black"
+            fi
+        fi
+    fi
+    return 0
+}
+
+# Checks whether phobos daemon is started. If not, tries to start it.
+# If start fails, raises an error.
+get_phobosd() {
+    save_log get_phobosd "BEGIN get_phobosd($*)"
+    op_ph=$(type -P phobos)
+    phd_ping_script="sudo $op_ph ping phobosd"
+    phd_ping=$($phd_ping_script >> $LOGFILE_FANCY 2>&1)
+    error_8=$?
+    save_log get_phobosd "$phd_ping_script return with status=$phd_ping($error_8)"
+    if [ $error_8 != 0 ]; then
+        # The ping failed. Try to start phobos daemon
+        phd_start_script="sudo systemctl start phobosd"
+        phd_start=$($phd_start_script)
+        error_8=$?
+        save_log get_phobosd "$phd_start_script return with status=$phd_start($error_8)"
+        if [ $error_8 == 0 ]; then
+            # Try to ping again phobosd after the start attempt
+            phd_ping_script="sudo $op_ph ping phobosd"
+            phd_ping=$($phd_ping_script >> $LOGFILE_FANCY 2>&1)
+            error_8=$?
+            save_log get_phobosd "$phd_ping_script return with status=$phd_ping($error_8)"
+        else
+            save_log get_phobosd "Impossible to communicate with Phobos daemon. Please check your Phobos and iRODS installation. Check that irods user has plain access to root privileges."
+            exit $error_8
+        fi
+    fi
+
+    if [ "$error_8" != 0 ]; then
+        STATUS="FAILURE"
+        save_log get_phobosd "STATUS=$STATUS($error_8)"
+    fi
+    save_log get_phobosd "END get_phobosd($*)"
+    return $error_8
+}
+
+# Get metadata on file $1 using $(stat), parsing it according to univMSS template, and return the output as a hash table.
+get_metadata() {
```

Comment:
question: in this case, what's the difference between metadata and stats?

Reply:
in this case, no difference that matters, I think.
This function could be renamed `get_status` considering `stat` "display file [...] status" (c.f. the manual page of `stat`). Though, to me, the set of information displayed is more about metadata than about status of a file.